src/index.d.ts

```diff
@@ -0,0 +1,55 @@
+type Selector = (strings: TemplateStringsArray, ... values: (string | number)[]) => string;
```

Comment:
value could be also a react component, like here https://github.com/lttb/reselector#babel

Reply:
and actually value is never primitive, it's object under the hood

I think it could be something like

```ts
type Selector = (strings: TemplateStringsArray, ... values: (ReactComponent<any> | ResolvedValue)[]) => string;
```